### ISSUE TITLE
onAddChild and new buffer logic

### ISSUE THREAD

**Issue by Ky6uk** [marionettejs/backbone.marionette#2394]
After upgrading from 2.3 to 2.4 I took some problem in using onAddChild method of CollectionView.

Follow code is not working now
```js
onAddChild: function (childView) {
    var view = new Marionette.ItemView;
    view.render();

    childView.$el.after(view.el); // node is not attached :(
}
```

That code not working only if ```isBuffering``` set to true. So `onAddChild` method a bit useless now.

**Comment by Ky6uk** [comment]
[JSBin example](http://jsbin.com/tihopa/2/edit?html,js,output)

**Comment by jasonLaster** [comment]
Just reproduced

Here's the code in question
```js
  onAddChild: function (childView) {
    childView.el.className = 'item';
    var newElement = $('<div>').addClass('item').html('test');
    childView.$el.after(newElement);  // works in 2.3, not in 2.4
  }
```

### 2.3 output (correct)
![](http://f.cl.ly/items/2Y3u0u0c2Q0J2i1n3Z1F/Image%202015-03-04%20at%2011.18.20%20AM.png)

2.4 doesn't have the test fields

**Comment by ahumphreys87** [comment]
The problem here is that during buffering the elements ain't on the page, we need to provide a way to reliably alter the output of the buffered HTML. This is a similar problem to the one @sentientnebula describes in #2388. Closing as a duplicate

**Comment by Ky6uk** [comment]
Looks like that problem won't be fixed in 2.4.2.

**Comment by theseyi** [comment]
Is there a tracking issue to fix this in 2.4.x or 3? Changing _createBuffer to createBuffer so it's not marked just for internal use would be a viable solution as suggested by @sentientnebula, @ahumphreys87

**Comment by Ky6uk** [comment]
@theseyi 2.4.2 fixed this issue.
